### PR TITLE
Resolve UIDs with host

### PR DESCRIPTION
**Pull request by tomgross**

### CODE REVIEW

File: plone/outputfilters/filters/resolveuid_and_caption.py [hunk 1]
```diff
@@ -333,18 +334,22 @@ def unknown_starttag(self, tag, attrs):
             if (tag == 'a' or tag == 'area') and 'href' in attributes:
                 href = attributes['href']
                 scheme = urlsplit(href)[0]
```
Comment (jensens) [inline plone/outputfilters/filters/resolveuid_and_caption.py]: ``scheme`` is redefined two lines below, so I think this line can be removed.

Comment (tomgross) [reply plone/outputfilters/filters/resolveuid_and_caption.py]: Fixed.

**Comment by thet** [comment]
This PR looks good, but I'm not sure about the problem this PR solves. The test case isn't very meaningful either. Can you elaborate a bit more?

File: plone/outputfilters/filters/resolveuid_and_caption.py [hunk 2]
```diff
@@ -332,19 +333,22 @@ def unknown_starttag(self, tag, attrs):
...
             if (tag == 'a' or tag == 'area') and 'href' in attributes:
                 href = attributes['href']
-                scheme = urlsplit(href)[0]
-                if not scheme \
-                        and not href.startswith('mailto<') \
+                url_parts = urlsplit(href)
+                scheme = url_parts[0]
+                # we are only interested in path and beyond /foo/bar?x=2#abc
+                path_parts = urlunsplit(['', ''] + list(url_parts[2:]))
+                if not href.startswith('mailto<') \
```
Comment (jensens) [inline plone/outputfilters/filters/resolveuid_and_caption.py]: ```mailto<``` does appear in which case?

**Comment by tomgross** [comment]
@thet The use case is: If you copy&paste relative (resolveuid) links in TinyMCE they are inserted as absolute links. If you have a multi domain setup (like in our setup we have webcms... for editors and www... for public consumtion but it can be different language domains www.de...., www.en.... too) the links are not resolved in all domains. This is fixed with this PR by just considering the path part for resolveuid pattern matching.

**Pull request merged**